File: MorySnip/Classes/Layers/Layer_Line.vb
Public Class Layer_Line
    Inherits Layer

    Public Sub New()

    End Sub

    Public Sub New(Pen As Pen, FirstPoint As Point)
        Me.Pen = Pen
        Me.Start(FirstPoint)
    End Sub

    Public Overrides Sub Paint(g As Graphics)
        g.DrawLine(Pen, FirstPoint, LastPoint)
    End Sub
End Class

File: MorySnip/Classes/Layers/LayerLine.cs
using System;
using System.Drawing;

namespace Morysoft.MorySnip.Classes.Layers;

public class LayerLine : Layer
{
    public LayerLine()
    {
    }

    public LayerLine(Pen pen, Point firstPoint)
    {
        this.Pen = pen;
        this.Start(firstPoint);
    }

    public float Length => (float)Math.Sqrt(
        Math.Pow(this.FirstPoint.X - this.LastPoint.X, 2) + Math.Pow(this.FirstPoint.Y - this.LastPoint.Y, 2)
    );

    public override void Paint(Graphics g) => g.DrawLine(this.Pen, this.FirstPoint, this.LastPoint);
}
